### ISSUE TITLE
Add ability to toggle status between Onlkine, Idle, DND, Invisble and Custom

### ISSUE THREAD

**Issue by Edstub207** [fredemmott/StreamDeck-Discord#110]
**Is your feature request related to a problem? Please describe.**
I would like to have buttons which toggle between Online, Idle, DND, Invisible and Custom defined options from my stream deck

**Describe the solution you'd like**
Option to add a new button type with pre-defined "online", "idle", "dnd", "Invisble" and also a custom status update.

**Comment by fredemmott** [comment]
I don't believe this is supported by Discord; the local RPC has been 'discouraged' for years (without a replacement), and most new features aren't there.

I took a look through and didn't see it, but if you'd like to double-check, https://discord.com/developers/docs/topics/rpc is the reference - though the named pipe/unix socket connection method now appears to be entirely undocumented.

**Comment by Edstub207** [comment]
Yeah, I did have a brief look before entering the task and thought I was on the RPC Docs, turns out I wasn't and it's only on Gateway. https://discord.com/developers/docs/topics/gateway#presence-update
